### PR TITLE
Generate API docs

### PR DESCRIPTION
**Pull request by SchrodingersGat**
Working towards external generation of API docs.

A *BIG* part of this PR will be removing our custom `filter_queryset` methods which do secret filtering that does not get exposed to the autodocs.

Ref: https://github.com/inventree/InvenTree/pull/6435

### CODE REVIEW

**Comment by matmair** [comment]
Maybe related to https://github.com/inventree/InvenTree/issues/6269?

**Comment by matmair** [comment]
grafik
7 unique errors left to go!

**Comment by matmair** [comment]
@SchrodingersGat all errors are gone but there is still a bunch of warnings that require separating reference names for enums and operations. How set are you on solving them?

**Comment by matmair** [comment]
@SchrodingersGat are you aware of `invoke schema`?

File: InvenTree/InvenTree/api.py [hunk 1]
```diff
@@ -324,7 +357,17 @@ def perform_create(self, serializer):
...
 
-class APISearchView(APIView):
+class APISearchViewSerializer(serializers.Serializer):
+    """Serializer for the APISearchView."""
+
+    search = serializers.CharField()
+    search_regex = serializers.BooleanField(default=False, required=False)
+    search_whole = serializers.BooleanField(default=False, required=False)
+    limit = serializers.IntegerField(default=1, required=False)
+    offset = serializers.IntegerField(default=0, required=False)
```
[the remark this answers is not in the document]
Comment (SchrodingersGat) [reply InvenTree/InvenTree/api.py:370]: I'm happy to leave as-is for now. We can always update later on

**Comment by SchrodingersGat** [comment]
> @SchrodingersGat are you aware of `invoke schema`?

I am now :) There are so many features now, I had forgotten about that. I  have now removed the duplicate task

**Comment by SchrodingersGat** [comment]
@matmair thanks for the help getting this one passing :)

**Pull request merged**